### ISSUE TITLE
Methods of SimpleHttp are after change now too much protected

### ISSUE THREAD

**Issue by dasniko** [keycloak/keycloak#30302]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

core

### Describe the bug

With #27656 the `do...()` methods of `SimpleHttp` were refactored by @ahus1 . I generally welcome this change, but...

The `do..()` methods with the `KeycloakSession` as second parameter are still public, the ones with the `HttpClient` are now protected. 
And there are no public `doPut()`, `doPatch()`, and `doHead()` methods with arguments `(String url, KeycloakSession session)`. Before the change, we could use the methods with the `HttpClient` for these methods. Now this is no more possible when using `SimpleHttp` as a good and simple http client inside of Keycloak extensions.
The use of `SimpleHttp` is the only easy, simple method to use a http client, as Keycloak doesn't provide other easy possibilities. Or does is meanwhile?

### Version

25.0.0

### Regression

- [X] The issue is a regression

### Expected behavior

Provide public static `SimpleHttp.do...()` methods for all http methods by default.

### Actual behavior

Only a few http methods are reflected by the public `SimpleHttp.do...()` methods.

### How to Reproduce?

Try to use the `SimpleHttp` class in a custom extension and try to use the `SimpleHttp.doPut(...)` method.

**Comment by ahus1** [comment]
Thanks for reaching out, and I'm sorry that this broke your code. 

This code is in `server-spi-private`, and there is no public API that we keep stable across releases. 

I seems to have missed that only some for some HTTP verbs there are public methods, and not for others. 

Can you provide an example where you prefer to call the doGet() without a `KeycloakSession` but with a `HttpClient` - maybe link to some code if your extension is public?

Otherwise, I'd rather add methods with `KeycloakSession` as they would restrict the response sizes. 

At the same time this resides in the package`org.keycloak.broker.provider.util` and it has never been designed to be public. The only API that was designed to be made public eventually was `HttpClientProvider`, but it only provides a subset of the functionality you are asking for. 

I'll leave it for the core team to triage this and decide which way to go.

**Comment by dasniko** [comment]
Thanks for answering quickly.
Perhaps I have expressed myself incorrectly. I **do not** want to methods with `httpClient` back, I just used them, e.g. `doPut` as there was no `doPut` with KeycloakSession.
So, basically, **I'd welcome** if for all the HTTP methods there would be a `doXXX(String url, KeycloakSession session)` method.

An yes, you are right, it's private, and internal, and and and... but Keycloak does really make it a _PITA_ to use an http client from within the custom extensions code - and doing http requests is kind of regular thing today. This should really be a lot more easier than today! I saw horrible code from people trying to leverage http requests from within Keycloak, only because KC doesn't provide a proper and convenient way of doing this.

(And I also can remember a comment from Stian some years ago, where he proposed to use the `SimpleHttp` class than something custom-crafted)
